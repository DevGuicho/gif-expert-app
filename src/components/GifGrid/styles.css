.grid__gifs {
  column-count: 4;
  column-gap: 10px;
}
.text__category {
  font-size: 20px;
  font-weight: bold;
  line-height: 30px;
}
@media screen and (max-width: 960px) {
  .grid__gifs {
    column-count: 3;
  }
}
@media screen and (max-width: 768px) {
  .grid__gifs {
    column-count: 2;
  }
}

@media screen and (max-width: 414px) {
  .grid__gifs {
    column-count: 1;
  }
}
