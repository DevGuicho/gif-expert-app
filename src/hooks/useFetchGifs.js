import { useEffect, useState } from 'react';
import getGifs from '../utilities/getGifs';

const useFetchGifs = ({ query = 'Dragon Ball Z' }) => {
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(false);
  const [gifs, setGifs] = useState([]);

  useEffect(() => {
    getGifs({ query: encodeURI(query) })
      .then((gifs) => {
        setGifs(gifs);
        setIsLoading(false);
      })
      .catch(() => setError(false));
  }, [query]);

  return { isLoading, error, gifs };
};

export default useFetchGifs;
