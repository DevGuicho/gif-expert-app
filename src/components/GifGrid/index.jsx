import React from 'react';
import PropTypes from 'prop-types';

import useFetchGifs from '../../hooks/useFetchGifs';
import GifGridItem from '../GifGridItem';
import './styles.css';

const GifGrid = ({ category }) => {
  const { gifs, isLoading, error } = useFetchGifs({ query: category });

  if (isLoading) return <h1>Loading...</h1>;

  if (error) return <h1>Hubo un error :(</h1>;

  return (
    <div>
      <h3 className="text__category">{category}</h3>
      <div className="grid__gifs">
        {gifs.map((gif) => (
          <GifGridItem key={gif.id} {...gif} />
        ))}
      </div>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
