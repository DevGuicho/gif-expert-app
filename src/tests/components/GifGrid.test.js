import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
  test('Debe mostrar correctamente el componente', () => {
    useFetchGifs.mockReturnValue({ isLoading: true, error: false, gifs: [] });

    const wrapper = shallow(<GifGrid category={'naruto'} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imagenes con useFetchGif', () => {
    const gifs = [{ id: '12', title: 'Naruto', url: 'https://localhost' }];
    useFetchGifs.mockReturnValue({
      isLoading: false,
      error: false,
      gifs,
    });
    const wrapper = shallow(<GifGrid category={'naruto'} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
