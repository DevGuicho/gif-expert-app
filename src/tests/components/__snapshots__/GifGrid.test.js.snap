// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar correctamente el componente 1`] = `
<h1>
  Loading...
</h1>
`;

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes con useFetchGif 1`] = `
<div>
  <h3
    className="text__category"
  >
    naruto
  </h3>
  <div
    className="grid__gifs"
  >
    <GifGridItem
      id="12"
      key="12"
      title="Naruto"
      url="https://localhost"
    />
  </div>
</div>
`;
